pkg/trojan/message.go

```diff
@@ -50,7 +36,13 @@ type Message struct {
 
 const (
 	// MaxPayloadSize is the maximum allowed payload size for the Message type, in bytes
+	// MaxPayloadSize + Topic + Length + Nonce = Default ChunkSize
+	//    (4030)      +  (32) +   (2)  +  (32) = 4096 Bytes
 	MaxPayloadSize = 4030
+	NonceSize      = 32
```

Comment:
nonce and topic are known to be hashes,  you can just use HashSize.
and better declare MaxPayloadSize transparently as:

     MaxPayloadSize = swarm.ChunkSize - NonceSize  - LengthSize - TopicSize

Reply:
@zelig , same comment, same time (second time) :D